pkg/storage/streams/store.go

```diff
@@ -352,7 +377,27 @@ func (s *streamStore) List(ctx context.Context, prefix, startAfter, endBefore pa
 		metaFlags |= meta.UserDefined
 	}
 
-	segments, more, err := s.segments.List(ctx, prefix.Prepend("l"), startAfter, endBefore, recursive, limit, metaFlags)
+	encPrefix, err := prefix.EncryptWithBucket(s.rootKey)
+	if err != nil {
+		return nil, false, err
+	}
+
+	prefixNoBucket := paths.Path([]string(prefix)[1:])
+	prefixKey, err := prefixNoBucket.DeriveKey(s.rootKey, len(prefixNoBucket))
+	if err != nil {
+		return nil, false, err
+	}
+
+	encStartAfter, err := startAfter.Encrypt(prefixKey)
```

Comment:
What if the prefix is `nil`? This would mean that `startAfter` and `endBefore` include the bucket part too, doesn't it? In this case, I guess we should call `EncryptWithBucket` instead.

Reply:
The problem is they won't include the bucket part if the prefix is not nil, so I will make two separate cases for this

Reply:
The new `listNoPrefix` helper method duplicates quite some logic from the main `List` method. It would be cleaner and easier to maintain if we would just define a smaller helper methods like:

```go
// encryptMarker is a helper method for encrypting startAfter and endBefore markers
func (s *streamStore) encryptMarker(marker paths.Path, prefixKey []byte) (paths.Path, error) {
	if bytes.Equal(s.rootKey, prefixKey) { // empty prefix
		return encryptAfterBucket(marker, s.rootKey)
	}
	return marker.Encrypt(prefixKey)
}
```

and use it to encrypt `startAfter` and `endBefore`.

Similar for decrypting.